lib/types/swagger_1.js

```diff
@@ -85,8 +85,10 @@ Swagger1.prototype.resolveSubResource = function (resource) {
     if (this.spec.basePath)
       baseUrl = URI(this.spec.basePath).absoluteTo(baseUrl);
 
-    if (this.getFormatVersion() === '1.0' && baseUrl.suffix() === 'json')
+    if (baseUrl.suffix() === 'json') {
       baseUrl.filename('');
+      if (resource.indexOf('.json') === -1) resource += '.json';
+    }
```

Comment:
The tests serve the test/input/ directory statically, and right now swagger_1 resolves the subresources as test/input/swagger/petstore/index.json/pet instead of test/input/swagger/petstore/pet.json